package/kedro_viz/integrations/kedro/sqlite_store.py

```diff
@@ -64,6 +87,69 @@ def save(self):
         engine, session_class = create_db_engine(self.location)
         Base.metadata.create_all(bind=engine)
         database = next(get_db(session_class))
-        session_store_data = RunModel(id=self._session_id, blob=self.to_json())
+
+        session_store_data = RunModel(id=self._session_id, blob=self._to_json())
         database.add(session_store_data)
         database.commit()
+        self.sync()
+
+    def _upload(self):
+        """Upload the session store db to s3"""
+        protocol, _ = get_protocol_and_path(self._remote_path)
+        db_name = _get_dbname()
+        fs = fsspec.filesystem(protocol)
+        with open(self.location, "rb") as file:
+            with fs.open(f"{self._remote_path}/{db_name}", "wb") as s3f:
+                s3f.write(file.read())
+
+    def _download(self) -> List[str]:
+        """Download all the dbs from an s3 bucket"""
+        protocol, _ = get_protocol_and_path(self._remote_path)
+        fs = fsspec.filesystem(protocol)
+        databases = fs.glob(f"{self._remote_path}/*.db")
+        databases_location = []
+        for database in databases:
+            database_name = os.path.basename(database)
+            with fs.open(database, "rb") as file:
+                db_data = file.read()
+            db_loc = f"{self._path}/{database_name}"
+            with open(db_loc, "wb") as temp_db:
+                temp_db.write(db_data)
+            databases_location.append(db_loc)
+        return databases_location
+
+    def _merge(self, databases_location: List[str]):
+        "Merge all dbs to the local session_store.db"
+        engine, session_class = create_db_engine(self.location)
+        Base.metadata.create_all(bind=engine)
+        database = next(get_db(session_class))
+
+        temp_engine = None
+        for db_loc in databases_location:
+            if temp_engine:
+                temp_engine.dispose()
+            temp_engine = create_engine(f"sqlite:///{db_loc}")
+            with temp_engine.connect() as database_conn:
+                db_metadata = MetaData()
+                db_metadata.reflect(bind=temp_engine)
+                for table_name, table_obj in db_metadata.tables.items():
+                    if table_name == "runs":
+                        data = database_conn.execute(table_obj.select()).fetchall()
+                        for row in data:
+                            row_dict = row._asdict()
+                            session_store_data = RunModel(**row_dict)
+                            try: 
+                                database.add(session_store_data)
+                                database.commit()
+                            except Exception:
+                                database.rollback()
```

Comment:
Do we need to add an exception message here, so if something goes wrong it's easier to debug?